### PR TITLE
New way to compile to libdevice calls for f64

### PR DESCRIPTION
**Pull request by fdrocha**
This is partly to generate cleaner triton code,
and partly to workaround a triton bug.

See https://github.com/pytorch/torchdynamo/pull/1605

### CODE REVIEW

File: torchinductor/lowering.py [hunk 1]
```diff
@@ -3204,25 +3212,33 @@ def sum_(x, axis=None, keepdims=False, *, dtype=None):
...
+    use_libdevice_for_f64=True,
 )
 register_pointwise(
-    aten.sin, type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT
+    aten.sin,
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
+    use_libdevice_for_f64=True,
 )
-register_pointwise(aten.abs)
+register_pointwise(aten.abs, use_libdevice_for_f64=True)
```
Comment (ngimel) [inline torchinductor/lowering.py:3241]: Can you please check if `abs` requires libdevice at all? I tried it at some point and seems like `tl.abs` accepts fp64 args.

**Review by ngimel — APPROVED** [review]
This looks good, I left minor comment about `abs`, can you please check if some of the functions on the skip list in `test_torchinductor_opinfo` can now pass or can be moved to xfail list because they no longer segfault?

**Comment by ngimel** [comment]
silu error looks real.

File: torchinductor/lowering.py (hunk 1 above)
Comment (fdrocha) [reply torchinductor/lowering.py:3241]: You are right! Will remove the override for `abs`.

**Comment by fdrocha** [comment]
> silu error looks real.

silu error happens because both `silu` and `sigmoid` are defined in `codegen/common.py` and they end up bypassing the overrides and dispatching to `tl.exp` even for `float64`.

I feel like the cleaner way to fix this is to make both `silu` and `sigmoid` come from primtorch decompositions instead, unless there is some downside in terms of compilation efficiency or something like that.

**Comment by fdrocha** [comment]
FYI, the triton issue is already fixed in master, see https://github.com/openai/triton/issues/773

We no longer need this change but still feels worth making because it leads to cleaner generated triton.

**Comment by fdrocha** [comment]
@ngimel could you please have a look at last commit?
I want to benchmark some of the changes I made, but I wanted to check if it passes CI first.

**Comment by jansel** [comment]
We have migrated `torchdynamo` to `torch._dynamo` and will use the `pytorch/pytorch` repo for future development.  Please resubmit this PR to https://github.com/pytorch/pytorch/

More details and instructions to port this PR over can be found in https://github.com/pytorch/torchdynamo/issues/1588

**Pull request closed without merging**